src/pages/pattern/pattern-details/pattern-details.js

```diff
@@ -0,0 +1,288 @@
+    setSmall,
+    thumbnail,
+    setThumbnail,
+    enName,
+    setEnName,
+    ukName,
+    setUkName,
+    setAvailable,
+    ukDescription,
+    setUkDescription,
+    enDescription,
+    setEnDescription,
+    material,
+    setMaterial,
+    handmade,
+    setHandmade,
+    available
+  } = usePatternHandlers();
+
+  useEffect(() => {
+    dispatch(getPattern(id));
+  }, [dispatch, id]);
+
+  useEffect(() => {
+    if (pattern !== null) {
+      setLarge(pattern.images.large);
+      setMedium(pattern.images.medium);
+      setSmall(pattern.images.small);
+      setThumbnail(pattern.images.thumbnail);
+      setUkName(pattern.name[0].value);
+      setEnName(pattern.name[1].value);
+      setUkDescription(pattern.description[0].value);
+      setEnDescription(pattern.description[1].value);
+      setHandmade(pattern.handmade);
+      setAvailable(pattern.available);
+      setMaterial(pattern.material);
+    }
+  }, [
+    pattern,
+    setUkName,
+    setEnName,
+    setUkDescription,
+    setEnDescription,
+    setAvailable,
+    setMaterial,
+    setHandmade,
+    setLarge,
+    setMedium,
+    setSmall,
+    setThumbnail
+  ]);
+
+  const patternSaveHandler = () => {
+    const pattern = {
+      name: [
+        {
+          lang: languages[0],
+          value: ukName
+        },
+        {
+          lang: languages[1],
+          value: enName
+        }
+      ],
+      description: [
+        {
+          lang: languages[0],
+          value: ukDescription
+        },
+        {
+          lang: languages[1],
+          value: enDescription
+        }
+      ],
+      images: {
+        large,
+        medium,
+        small,
+        thumbnail
+      },
+      material,
+      handmade,
+      available
+    };
+    dispatch(updatePattern({ id, pattern }));
+  };
+
+  if (loading) {
+    return <LoadingBar />;
+  }
+  const checkboxes = [
+    {
+      id: 'handmade',
+      value: handmade,
+      checked: handmade,
+      color: 'primary',
+      label: 'зроблений вручну',
+      handler: (e) => setHandmade(e.target.checked)
+    },
+    {
+      id: 'available',
+      value: available,
+      checked: available,
+      color: 'primary',
+      label: 'доступний',
+      handler: (e) => setAvailable(e.target.checked)
+    }
+  ];
+
+  const checkboxInputs = checkboxes.map(
+    ({ color, checked, label, value, handler }) => (
+      <FormControlLabel
+        key={label}
+        value={value}
+        checked={checked}
+        control={<Checkbox color={color} />}
+        label={label}
+        labelPlacement='start'
+        onChange={handler}
+      />
+    )
+  );
+
+  return (
+    <div className={classes.detailsContainer}>
+      <form className={classes.form} onSubmit={patternSaveHandler}>
+        <FormControl className={classes.patternDetails}>
+          <Grid container spacing={1}>
+            {checkboxInputs}
+            <Grid item xs={12}>
+              <Paper className={classes.patternItemUpdate}>
```

Comment:
Please, move this component `Paper` and its inner components to the separate component to reduce size of current file and improve readability

Reply:
refactored

Reply:
fixed